src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -230,6 +231,65 @@ master_drop_all_shards(PG_FUNCTION_ARGS)
 
 
 /*
+ * master_drop_sequences attempts to drop a list of sequences on a specified
+ * node. The "IF EXISTS" clause is used to permit dropping sequences even if
+ * they may not exist. Returns true on success, false on failure.
+ */
+Datum
+master_drop_sequences(PG_FUNCTION_ARGS)
+{
+	ArrayType *sequenceNamesArray = PG_GETARG_ARRAYTYPE_P(0);
+	text *nodeText = PG_GETARG_TEXT_P(1);
+	int64 nodePort = PG_GETARG_INT64(2);
+	bool dropSuccessful = false;
+	char *nodeName = TextDatumGetCString(nodeText);
+
+	ArrayIterator sequenceIterator = NULL;
+	Datum sequenceText = 0;
+	bool isNull = false;
+
+	StringInfo dropSeqCommand = makeStringInfo();
+
+	/* this runs from a trigger that must run as superuser */
+	if (!superuser())
```

Comment:
Well, just because one place does something suboptimal, doesn't mean we should introduce more of the same.  I'm ok with either just doing it right here, or fixing both.